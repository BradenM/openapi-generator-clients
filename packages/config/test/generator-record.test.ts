import {expect, it} from 'vitest'
import {
	GeneratorRecord,
	OASGeneratorRecord,
	OASRecord,
	TemplateConfigRecord,
} from '../src'

it.each`
	rec
	${GeneratorRecord}
	${TemplateConfigRecord}
	${OASGeneratorRecord}
	${OASRecord}
`('$rec.displayName inits with defaults', async ({rec}) => {
	const inst = rec()
	expect(inst.toJS()).toMatchSnapshot(rec.displayName)
})

it('OASRecord from GeneratorRecord', async (ctx) => {
	const rec = GeneratorRecord({
		drop: ['a', 'b'],
	})
	console.log(rec.toJS())
	ctx.expect(rec.toJS()).toMatchInlineSnapshot(`
    {
      "additionalArgs": [],
      "additionalProperties": {
        "disallowAdditionalPropertiesIfNotPresent": undefined,
        "legacyDiscriminatorBehavior": false,
        "npmName": undefined,
        "platform": undefined,
        "supportsES6": true,
        "useInversify": undefined,
        "useObjectParameters": undefined,
        "usePromise": true,
        "useRxJS6": true,
        "withInterfaces": true,
      },
      "drop": [
        "a",
        "b",
      ],
      "generatorName": "",
      "generatorVersion": "",
      "gitHost": "github.com",
      "gitRepoId": "",
      "gitUserId": "",
      "glob": undefined,
      "inputSpec": undefined,
      "legacyDiscriminatorBehavior": false,
      "output": "",
      "removeOperationIdPrefix": true,
      "templateDir": "",
    }
  `)
	const oas = OASRecord({
		'generator-cli': {
			generators: {
				mygenerator: OASGeneratorRecord(rec),
			},
		},
	})
	ctx.expect(oas.toJS()).toMatchInlineSnapshot(`
    {
      "$schema": "node_modules/@openapitools/openapi-generator-cli/config.schema.json",
      "generator-cli": {
        "generators": {
          "mygenerator": {
            "additionalProperties": {
              "disallowAdditionalPropertiesIfNotPresent": undefined,
              "legacyDiscriminatorBehavior": false,
              "npmName": undefined,
              "platform": undefined,
              "supportsES6": true,
              "useInversify": undefined,
              "useObjectParameters": undefined,
              "usePromise": true,
              "useRxJS6": true,
              "withInterfaces": true,
            },
            "generatorName": "",
            "gitHost": "github.com",
            "gitRepoId": "",
            "gitUserId": "",
            "glob": undefined,
            "inputSpec": undefined,
            "legacyDiscriminatorBehavior": false,
            "output": "",
            "removeOperationIdPrefix": true,
            "templateDir": "",
          },
        },
      },
      "spaces": 2,
    }
  `)
})
