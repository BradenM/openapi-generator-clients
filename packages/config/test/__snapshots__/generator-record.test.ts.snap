// Vitest Snapshot v1

exports[`'GeneratorRecord' inits with defaults > GeneratorRecord 1`] = `
{
  "additionalArgs": [],
  "additionalProperties": {
    "disallowAdditionalPropertiesIfNotPresent": undefined,
    "legacyDiscriminatorBehavior": false,
    "npmName": undefined,
    "platform": undefined,
    "supportsES6": true,
    "useInversify": undefined,
    "useObjectParameters": undefined,
    "usePromise": true,
    "useRxJS6": true,
    "withInterfaces": true,
  },
  "drop": [],
  "files": {},
  "generatorName": "",
  "generatorVersion": "",
  "gitHost": "github.com",
  "gitRepoId": "",
  "gitUserId": "",
  "glob": undefined,
  "inputSpec": undefined,
  "legacyDiscriminatorBehavior": false,
  "output": "",
  "removeOperationIdPrefix": true,
  "templateDir": "",
}
`;

exports[`'OASGeneratorRecord' inits with defaults > OASGeneratorRecord 1`] = `
{
  "additionalProperties": {
    "disallowAdditionalPropertiesIfNotPresent": undefined,
    "legacyDiscriminatorBehavior": false,
    "npmName": undefined,
    "platform": undefined,
    "supportsES6": true,
    "useInversify": undefined,
    "useObjectParameters": undefined,
    "usePromise": true,
    "useRxJS6": true,
    "withInterfaces": true,
  },
  "files": {},
  "generatorName": "",
  "gitHost": "github.com",
  "gitRepoId": "",
  "gitUserId": "",
  "glob": undefined,
  "inputSpec": undefined,
  "legacyDiscriminatorBehavior": false,
  "output": "",
  "removeOperationIdPrefix": true,
  "templateDir": "",
}
`;

exports[`'OASRecord' inits with defaults > OASRecord 1`] = `
{
  "$schema": "node_modules/@openapitools/openapi-generator-cli/config.schema.json",
  "generator-cli": {
    "generators": {},
    "version": "6.0.0-beta",
  },
  "spaces": 2,
}
`;

exports[`'TemplateConfigRecord' inits with defaults > TemplateConfigRecord 1`] = `
{
  "additionalArgs": [],
  "additionalProperties": {
    "disallowAdditionalPropertiesIfNotPresent": undefined,
    "legacyDiscriminatorBehavior": false,
    "npmName": undefined,
    "platform": undefined,
    "supportsES6": true,
    "useInversify": undefined,
    "useObjectParameters": undefined,
    "usePromise": true,
    "useRxJS6": true,
    "withInterfaces": true,
  },
  "drop": [],
  "files": {},
  "generatorVersion": "",
}
`;
